google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UpdateTargetingSetting.java

```diff
@@ -0,0 +1,191 @@
+   * @param customerId the client customer ID.
+   * @param adGroupId the ID of the ad group to update.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long adGroupId) {
+    // Creates an empty TargetingSetting object.
+    TargetingSetting.Builder targetingSettingBuilder = TargetingSetting.newBuilder();
+
+    // Creates the Google Ads service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Creates a search request that retrieves the targeting settings from a given ad group.
+      String searchQuery =
+          "SELECT ad_group.id, ad_group.name, ad_group.targeting_setting.target_restrictions FROM"
+              + " ad_group WHERE ad_group.id = "
+              + adGroupId;
+
+      // Creates a request that will retrieve all ad groups using pages of the specified page size.
+      SearchGoogleAdsRequest request =
+          SearchGoogleAdsRequest.newBuilder()
+              .setCustomerId(Long.toString(customerId))
+              .setPageSize(PAGE_SIZE)
+              .setQuery(searchQuery)
+              .build();
+      // Issues the search request.
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Iterates over all rows in all pages and prints the requested field values for the ad group
+      // in each row.
+      for (GoogleAdsRow googleAdsRow : searchPagedResponse.iterateAll()) {
+        AdGroup adGroup = googleAdsRow.getAdGroup();
+        // Prints the results.
+        System.out.printf(
+            "Ad group with ID %d and name '%s' was found with the following targeting"
+                + " restrictions.%n",
+            adGroup.getId().getValue(), adGroup.getName().getValue());
+        List<TargetRestriction> targetRestrictions =
+            adGroup.getTargetingSetting().getTargetRestrictionsList();
+        // Loops through and prints each of the target restrictions.
+        // Reconstructs the TargetingSetting object with the updated audience target restriction
+        // because Google will overwrite the entire targeting_setting field of the ad group when
+        // the field mask includes targeting_setting in an update operation.
+        for (TargetRestriction targetRestriction : targetRestrictions) {
+          TargetingDimension targetingDimension = targetRestriction.getTargetingDimension();
+          boolean bidOnly = targetRestriction.getBidOnly().getValue();
+          System.out.printf(
+              "  Targeting restriction with targeting dimension '%s' and bid only set to '%b'.%n",
+              targetingDimension, bidOnly);
+          // Adds the target restriction to the TargetingSetting object as is if the targeting
+          // dimension has a value other than AUDIENCE because those should not change.
```

Comment:
Can you reword this sentence. It's a bit hard to understand.
The explanation at lines 136-139 is much easier to comprehend.